gobblin-modules/gobblin-http/src/main/java/gobblin/r2/R2RestResponseHandler.java

```diff
@@ -27,6 +29,7 @@
 
   public static final String CONTENT_TYPE_HEADER = "Content-Type";
   private final Set<String> errorCodeWhitelist;
+  MetricContext metricsContext = new MetricContext.Builder("R2ResponseStatus").build();
```

Comment:
If the context should be shared among a writer and its response handler, it can be passed in on initialization